include/flatbuffers/reflection.h

```diff
@@ -46,7 +46,9 @@ inline bool IsLong(reflection::BaseType t) {
 // Size of a basic type, don't use with structs.
 inline size_t GetTypeSize(reflection::BaseType base_type) {
   // This needs to correspond to the BaseType enum.
-  static size_t sizes[] = { 0, 1, 1, 1, 1, 2, 2, 4, 4, 8, 8, 4, 8, 4, 4, 4, 4 };
+  static size_t sizes[] = { 0, 1, 1, 1, 1, 2, 2, 4, 4, 8, 8, 4, 8, 4, 4, 4, 4, 0 };
```

Comment:
can we use sizeof() here instead of listing out a bunch of magic numbers?

Reply:
Valid question, but that is a separate issue from the one that I'm trying to address with this change.